src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Optional;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        if (fruitTransaction == null) {
+            throw new IllegalArgumentException("FruitTransaction cannot be null");
+        }
+        String fruit = fruitTransaction.getFruit();
+        if (fruit == null || fruit.isEmpty()) {
+            throw new IllegalArgumentException("Fruit cannot be null or empty");
+        }
+        int amountToPurchase = fruitTransaction.getQuantity();
+        if (amountToPurchase <= 0) {
+            throw new IllegalArgumentException("Quantity must be greater than zero");
+        }
+        Optional<Integer> currentBalance = Optional.ofNullable(Storage.fruitStorage.get(fruit));
+        int balanceAfterPurchase = currentBalance.orElse(0) - amountToPurchase;
+        if (balanceAfterPurchase != 0) {
```

Comment:
The condition `balanceAfterPurchase != 0` seems incorrect. You should be checking if `balanceAfterPurchase` is negative instead. If `balanceAfterPurchase` is zero, it means that all the stock for that fruit has been sold, which is a valid state. However, if it's negative, it indicates an attempt to purchase more than the available stock, which should not be allowed.

Reply:
agree, condition looks confusing and is imprecise, check if it's negative